department-of-veterans-affairs/va.gov-cms
# VACMS-6454: remove title from vet center migration - copy from field_official_name

## Description

closes #6454 

## Testing done

Visual testing

## Screenshots

Checked the content page `/admin/content` and filtered by Content type = Vet Center to see how many items there are:

<img width="1606" alt="Screen Shot 2021-10-15 at 1 50 00 PM" src="https://user-images.githubusercontent.com/21045418/137531184-2ab77115-5221-42be-b15c-459c45af0248.png">

From the terminal I then ran `lando drush mr va_node_facility_vet_centers` to wipe the migration map table

Next I ran the migration fresh with `lando drush migrate:import va_node_facility_vet_centers --update`

I then checked the content page `/admin/content` and filtered by Content type = Vet Center again to see the item count. Because the migration map table was wiped out we now have duplicates to test. I then searched for a specific Vet Center to be sure there were two copies and that the new one has a title:

<img width="1626" alt="Screen Shot 2021-10-15 at 1 58 34 PM" src="https://user-images.githubusercontent.com/21045418/137532161-352c6b01-073f-4cd9-9fdd-e090cb29201e.png">

I then edited the new item's title and ran the migration again. To edit the title you need to make the field visible (It is labeled: Name of Facility) here: `/admin/structure/types/manage/vet_center/form-display`. You will also need to comment out the va_gov_consumers_form_alter function on line 14 of `docroot/modules/custom/va_gov_consumers/va_gov_consumers.module`. Once the migration finishes running I checked the edited Vet Center to make sure the page title change I just made wasn't reverted.

<img width="1611" alt="Screen Shot 2021-10-15 at 2 41 59 PM" src="https://user-images.githubusercontent.com/21045418/137537477-ef604f9c-eb61-41fa-b141-2578761ecba4.png">

## QA steps

As an admin user
1. [x] Check the content page `/admin/content` and filter by Content type = Vet Center to see how many items there are
2. [x] From the terminal run `lando drush mr va_node_facility_vet_centers` to wipe the migration map table
3. [x] From the terminal run `lando drush migrate:import va_node_facility_vet_centers --update`
4. [x] Visit `/admin/structure/types/manage/vet_center/form-display` and set the **Name of Facility** field to visible
5. [x] In your IDE comment out the va_gov_consumers_form_alter() function in va_gov_consumers.module
6. [x] Check the content page `/admin/content` and search for a specific Vet Center (you should see two entries)
7. [x] Edit the most recently migrated copy and change the **Name of Facility** (note the new name)
8. [x] From the terminal run `lando drush migrate:import va_node_facility_vet_centers --update` again
9. [x] Refresh the `/admin/content` page and filter by the new name you created in step 7

If the new name you created was not overwritten by the migration running again then you're good to go.
